projects/dashboards/src/widgets/base.model.ts

```diff
@@ -16,6 +17,13 @@ export abstract class BaseModel {
   })
   public id?: string;
 
+  @ModelProperty({
+    key: 'noDataLoadingConfig',
+    required: false,
+    type: PLAIN_OBJECT_PROPERTY.type
+  })
+  public noDataLoadingConfig?: NoDataOrErrorStateConfig;
```

Comment:
Why only base `no data` part? We could also use `loadAsyncConfig` .
cc: @anandtiwary

Reply:
After talking about it we decided only `noData` should be configurable to the consumer of widgets. Error state is the same across the app for `loadAsyncConfig` so it should be controlled within the directive itself.  The widgets should be responsible for `loaderType`, so it should be non-configurable to widget consumers. Please correct if needed Anand